From: Kefeng Wang <wangkefeng.wang@huawei.com>
Subject: mm/damon/paddr: minor refactor of damon_pa_mark_accessed_or_deactivate()
Date: Wed, 8 Mar 2023 16:33:10 +0800

Omit one line by unified folio_put(), and make code more clear.

Link: https://lkml.kernel.org/r/20230308083311.120951-3-wangkefeng.wang@huawei.com
Signed-off-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Reviewed-by: SeongJae Park <sj@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/damon/paddr.c |    7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

--- a/mm/damon/paddr.c~mm-damon-paddr-minor-refactor-of-damon_pa_mark_accessed_or_deactivate
+++ a/mm/damon/paddr.c
@@ -268,16 +268,15 @@ static inline unsigned long damon_pa_mar
 		if (!folio)
 			continue;
 
-		if (damos_pa_filter_out(s, folio)) {
-			folio_put(folio);
-			continue;
-		}
+		if (damos_pa_filter_out(s, folio))
+			goto put_folio;
 
 		if (mark_accessed)
 			folio_mark_accessed(folio);
 		else
 			folio_deactivate(folio);
 		applied += folio_nr_pages(folio);
+put_folio:
 		folio_put(folio);
 	}
 	return applied * PAGE_SIZE;
_
